extensions/ui/org.eclipse.smarthome.ui.basic/src/main/java/org/eclipse/smarthome/ui/basic/internal/render/SelectionRenderer.java

```diff
@@ -146,6 +117,46 @@ private String getMappingsJSON(Selection w) {
         return null;
     }
 
+    private String buildRow(Selection w, String lab, String cmd, Item item, State state, StringBuilder rowSB)
+            throws RenderException {
+        String mappingLabel = null;
+        String rowSnippet = getSnippet("selection_row");
+
+        String command = cmd != null ? cmd : "";
+        String label = lab;
+
+        if (item instanceof NumberItem && ((NumberItem) item).getDimension() != null) {
+            String unit = getUnitForWidget(w);
+            command = StringUtils.replace(command, UnitUtils.UNIT_PLACEHOLDER, unit);
+            label = StringUtils.replace(label, UnitUtils.UNIT_PLACEHOLDER, unit);
+
+            // Special treatment for °C since uom library uses a single character: ℃
+            // This will ensure the current state matches the cmd and the buttonClass is set accordingly.
+            command = StringUtils.replace(command, "°C", "℃");
```

Comment:
Hm, this might break through https://github.com/eclipse/smarthome/issues/5669 - it would probably better to replace the special char and always compare with "°C" here (same for the Classic UI, I noticed that too late).

Reply:
One time again, this is only a move of the existing code. My PR is not fixing something relative to UoM. Maybe the fix should be applied globally in another PR if there is a problem with °Cellsius unit ?
If you prefer, I can rebase my PR after this problem is globally fixed ?

Reply:
Ok, I didn't see that this was indeed in there before, sorry.
So let's keep it here, we just have to make sure to remember to address those places once #5669 is addressed (cc @htreu).